app.ts

```diff
@@ -58,6 +79,38 @@ const fetchInstanceAndForm = async function (formId: string, id: string) {
   return { form, instance };
 };
 
+app.get('/:formId/instances', async function (req, res, next) {
+  const form = await fetchFormDefinitionById(req.params.formId);
+  if (!form) {
+    res.send(404);
+    return;
+  }
+
+  const formLabel = await getFormLabel(form.formTtl);
+  if (!formLabel) {
+    res.send(500);
+    return;
+  }
+
+  const getInstanceIdQuery = getFormInstancesQuery(formLabel);
+  const queryResult = await executeQuery(getInstanceIdQuery, next);
+
+  const instance_values: Instance[] = [];
+
+  queryResult.results.bindings.map((binding) => {
+    const instance = {
+      uri: binding.uri.value,
+      id: binding.id.value,
+      label: formLabel,
+    };
+    instance_values.push(instance);
+  });
```

Comment:
Ideally, I would keep the routing methods as small as possible

Reply:
I'll split it up, ideally we should a dedicated router and no logic would be present here, maybe something to discuss later.